IMS.BL/Domain/CustomExceptions/ProductException.cs

```diff
@@ -0,0 +1,13 @@
+using System;
+
+namespace IMS.BL.Domain.CustomExceptions
+{
+    public class ProductException : Exception
```

Comment:
you exception is still too generic, what does this product do, is it a ProductNotFoundException or is it a ProductAlreadyExistsException or some other domain specific case, ProductException is no different from exception still too generic, if you need to add more that one custom exception for the same entity go ahead and add more than one.

to check if your exception is generic one or not, add a default message for the case it handle, it you need to pass a custom message for a different case you want to handle then that means you need a new exception.